flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1358,4 +1358,19 @@ public void actionsUnavailable() throws InterruptedException {
 			mailbox.putMail(actionUnavailableLetter);
 		}
 	}
+
+	private class TimerInvocationContext implements SystemProcessingTimeService.ScheduledCallbackExecutionContext {
+		@Override
+		public void invoke(ProcessingTimeCallback callback, long timestamp) throws InterruptedException {
```

Comment:
Are you sure that it is a good idea to throw the `InterruptedException` here? It seems more like this is something that is currently hard to handle appropriately for the caller and should probably be handled as async exception as well (i.e. smehow stop the task).